lib/datadog/tracing/contrib/rack/middlewares.rb

```diff
@@ -91,7 +91,11 @@ def call(env)
             # we must ensure that the span `resource` is set later
             request_span = Tracing.trace(Ext::SPAN_REQUEST, **trace_options)
             request_span.resource = nil
-            request_trace = Tracing.active_trace
+
+            # When tracing and distributed tracing are both disabled, `.active_trace` will be `nil`,
+            # Return a null object to continue operation
+            request_trace = Tracing.active_trace || TraceOperation.new
```

Comment:
The exception happens in the call to `set_request_tags!` made in this method. Wouldn't it be better to expand the condition on line 72 to check if tracing is disabled, and if so, skip the remainder of the method?